### PR TITLE
Use asset manager store for organisation logos

### PR DESCRIPTION
**Pull request by chrislo**
Organisation logos are now being stored in and served from Asset
Manager, so there is no longer any need to store them on the local,
NFS filesystem. This PR switches the storage engine for
`LogoUploader` to `AssetManagerStorage`.

### CODE REVIEW

**Comment by floehopper** [comment]
The PR title and the subject line for the final commit is ambiguous - when I read it I thought the "only" referred to the organisation logos. Maybe something like "Use asset manager vs combined store for organisation logos" would be better...?

File: test/unit/whitehall/asset_manager_storage_test.rb [hunk 1]
```diff
@@ -53,7 +54,7 @@ class AssetManagerUploader < CarrierWave::Uploader::Base
     storage.retrieve!('identifier')
   end
 
-  test 'returns an asset manager file' do
+  test 'retrieve! returns an asset manager file' do
```
Comment (floehopper) [inline test/unit/whitehall/asset_manager_storage_test.rb]: Minor: Ideally this would be in a separate commit.

File: test/functional/organisations_controller_test.rb [hunk 2]
```diff
@@ -191,7 +191,7 @@ def self.sets_cache_control_max_age_to_time_of_next_scheduled(edition_type)
       organisation_logo_type_id: OrganisationLogoType::CustomLogo.id,
       logo: fixture_file_upload('logo.png')
     )
-    VirusScanHelpers.simulate_virus_scan(organisation.logo)
+
```
Comment (floehopper) [inline test/functional/organisations_controller_test.rb]: Minor: Would it be feasible to remove the calls to `VirusScanHelpers.simulate_virus_scan` in a separate commit?

File: lib/whitehall/asset_manager_storage.rb [hunk 3]
```diff
@@ -29,8 +29,7 @@ def delete
     end
 
     def url
-      asset = Services.asset_manager.whitehall_asset(@legacy_url_path)
-      asset['file_url']
+      URI.join(Plek.new.public_asset_host, @legacy_url_path).to_s
```
[the remark this answers is not in the document]
Comment (chrislo) [reply lib/whitehall/asset_manager_storage.rb]: `Plek#public_asset_host` is [only used](https://github.com/search?p=1&q=org%3Aalphagov+public_asset_host&type=Code) in Whitehall and Plek itself. I'm reasonably confident that if it needs to change the use of it introduced in this PR will be noticed.

File: test/unit/presenters/publishing_api/organisation_presenter_test.rb [hunk 4]
```diff
@@ -64,11 +64,11 @@ def present(model_instance, options = {})
       '/government/uploads/system/uploads/organisation/logo/1/960x640_jpeg.jpg'
 
     assert_equal(
-      presented_item.content[:details][:logo][:image],
       {
         url: expected_image_url,
         alt_text: 'Organisation of Things',
-      }
+      },
+      presented_item.content[:details][:logo][:image]
```
[the remark this answers is not in the document]
Comment (chrislo) [reply test/unit/presenters/publishing_api/organisation_presenter_test.rb]: I'll amend the commit note.

File: test/unit/whitehall/asset_manager_storage_test.rb [hunk 5]
```diff
@@ -75,10 +75,13 @@ class Whitehall::AssetManagerStorage::FileTest < ActiveSupport::TestCase
     @file.delete
   end
 
-  test 'uses the asset-manager api to return the url of the file' do
-    asset_url = 'http://asset-host/asset.png'
-    Services.asset_manager.stubs(:whitehall_asset).with(@asset_url_path).returns('file_url' => asset_url)
+  test 'constructs the url of the file using the public asset host and legacy url path' do
+    plek = stub('plek')
+    plek.stubs(:public_asset_host).returns('http://assets-host')
```
[the remark this answers is not in the document]
Comment (chrislo) [reply test/unit/whitehall/asset_manager_storage_test.rb]: I'll make both of those changes and fixup.

File: test/unit/whitehall/asset_manager_storage_test.rb (hunk 1 above)
Comment (chrislo) [reply test/unit/whitehall/asset_manager_storage_test.rb]: Done.

File: test/integration/asset_manager_test.rb [hunk 6]
```diff
@@ -27,6 +27,31 @@ class CreatingAnOrganisationLogo < ActiveSupport::TestCase
...
+        :consultation_response_form_data,
+        file: File.open(Rails.root.join('test', 'fixtures', @filename))
+      )
+    end
+
+    test 'sends the consultation response form data file to Asset Manager' do
+      Services.asset_manager.expects(:create_whitehall_asset).with do |args|
+        args[:file].is_a?(File) &&
+          args[:legacy_url_path] =~ /#{@filename}/
+      end
```
[the remark this answers is not in the document]
Comment (chrislo) [reply test/integration/asset_manager_test.rb]: These assertions are copied from the other tests in this file, so I'm happy to address your suggested refactoring against the whole file in a separate PR.

**Comment by chrislo** [comment]
Thanks @floehopper! Rebasing onto master and force-pushing before merging.
